### PR TITLE
[WIP][Fix] rubocop is now working

### PR DESCRIPTION
**Pull request by grahamjenson**

### CODE REVIEW

**Comment by kigster** [comment]
I thought we would create a new command `ruby_rubocop` for this, no?

File: ruby/private/binary_wrapper.tpl [hunk 1]
```diff
@@ -24,6 +19,7 @@
 #
 
 require 'rbconfig'
+require 'rubygems'
```
Comment (yugui) [inline ruby/private/binary_wrapper.tpl]: Why do you need this?

File: ruby/private/binary_wrapper.tpl [hunk 2]
```diff
@@ -122,28 +118,10 @@ def main(args)
       end
     end
   end
-  exec(ruby_program, '--disable-gems', *rubyopt, main, *args)
+  exec(ruby_program, *rubyopt, main, *args)
```
Comment (yugui) [inline ruby/private/binary_wrapper.tpl]: ?

File: ruby/private/bundle/create_bundle_build_file.rb [hunk 3]
```diff
@@ -78,18 +64,19 @@ def create_bundle_build_file(build_out_file, lock_file, repo_name, excludes, wor
   bundle.specs.each { |spec|
     deps = spec.dependencies.map(&:name)
     deps += [":bundler_setup"]
-    template_out += GEM_TEMPLATE.gsub("{exclude}", excludes)
+
+    exclude_array = excludes[spec.name] || []
+    # We want to exclude files and folder with spaces in them
+    exclude_array += ["**/* *.*", "**/* */*"]
+
+    template_out += GEM_TEMPLATE.gsub("{exclude}", exclude_array.to_s)
```
Comment (yugui) [inline ruby/private/bundle/create_bundle_build_file.rb]: It might be been simpler to use `String#%`.

File: examples/simple_script/BUILD [hunk 4]
```diff
@@ -13,6 +13,7 @@ ruby_binary(
     deps = [
         "//lib:foo",
         "@bundle//:awesome_print",
+        "@bundle//:rails",
```
Comment (yugui) [inline examples/simple_script/BUILD]: Is this change related to rubocop?

File: ruby/private/binary.bzl [hunk 5]
```diff
@@ -29,50 +29,27 @@ def _ruby_binary_impl(ctx):
             "main",
         )
 
-    if sdk.is_host:
```
Comment (yugui) [inline ruby/private/binary.bzl]: Although it is OK to remove this until we really start including runtime into ruby_binary, is it related to rubocop?

File: examples/simple_script/BUILD [hunk 6]
```diff
@@ -30,3 +31,28 @@ ruby_test(
         "@bundle//:rspec",
     ],
 )
+
+
+ruby_binary(
+    name = "rubocop",
+    main = "@bundle//:bin/rubocop",
```
Comment (kigster) [inline examples/simple_script/BUILD]: This is really cool — being able to reference binaries like that! ✅

Do you think it makes sense to create a standalone `ruby_rubocop` Bazel command?

File: ruby/private/binary_wrapper.tpl (hunk 1 above)
Comment (kigster) [reply ruby/private/binary_wrapper.tpl]: Probably because we disable rubygems when we run `ruby` interpreter.

@yugui  I was wondering what was the motivation behind `--disable-rubygems` flag?

File: ruby/private/binary_wrapper.tpl (hunk 2 above)
Comment (kigster) [reply ruby/private/binary_wrapper.tpl]: Don't we need access to RubyGems subsystem when running our ruby code?

**Comment by grahamjenson** [comment]
@yugui @kigster deprecated this for https://github.com/bazelruby/rules_ruby/pull/38

File: examples/simple_script/BUILD (hunk 6 above)
Comment (grahamjenson) [reply examples/simple_script/BUILD]: we will, but best get the basics working first

File: ruby/private/binary.bzl (hunk 5 above)
Comment (grahamjenson) [reply ruby/private/binary.bzl]: What ever we do I am not 100% sure we should include the runtime from the toolchain into the binary, and probably should reference another build object, e.g. a ruby_runtime.

File: examples/simple_script/BUILD (hunk 4 above)
Comment (grahamjenson) [reply examples/simple_script/BUILD]: this was a test branch, which I have cleaned up on https://github.com/bazelruby/rules_ruby/pull/38

File: ruby/private/bundle/create_bundle_build_file.rb (hunk 3 above)
Comment (grahamjenson) [reply ruby/private/bundle/create_bundle_build_file.rb]: I have cleaned up the excludes rules here https://github.com/bazelruby/rules_ruby/pull/38/files

File: ruby/private/binary_wrapper.tpl (hunk 2 above)
Comment (grahamjenson) [reply ruby/private/binary_wrapper.tpl]: this was specifically a problem with rubocop which requires the ability to inspect the gems it depends on, e.g.  https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/yaml_duplication_checker.rb

File: ruby/private/binary_wrapper.tpl (hunk 1 above)
Comment (grahamjenson) [reply ruby/private/binary_wrapper.tpl]: This is a mess I cleaned up in https://github.com/bazelruby/rules_ruby/pull/38/commits/e93b7b920274300c4582622a5bba30de19408f62
